### PR TITLE
Move CI Build_And_UnitTest to scale set agents

### PR DESCRIPTION
**Pull request by zivkan**
## Bug

Fixes: https://github.com/NuGet/Client.Engineering/issues/819

Regression? no

## Description

* Run build + unit test jobs on scale set agents
* Remove publish symbols for 45 days step
* Make changes for VSTS drop steps, to work on new infrastructure

## PR Checklist

- [X] PR has a meaningful title
- [X] PR has a linked issue.
- [X] Described changes

- **Tests**
  - [ ] Automated tests added
  - **OR**
  
  - [ ] Test exception
  - **OR**
  - [X] N/A: Infrastructure

- **Documentation**
  
  - [ ] Documentation PR or issue filled
  - **OR**
  - [X] N/A

### CODE REVIEW

File: eng/pipelines/templates/Build_and_UnitTest.yml
Comment (zivkan) [inline eng/pipelines/templates/Build_and_UnitTest.yml:478]: @nkolev92 @dtivel or anyone else,  do you know if there's a reason why we've been publishing symbols for 45 days on each build? As part of our insertion process, we already publish symbols (and not limited to 45 days), so I can't think of any reason this step is actually useful.

Comment (zivkan) [reply eng/pipelines/templates/Build_and_UnitTest.yml:478]: Actually, maybe it's useful for manual vendor testing and OptProf tests, so if VS crashes we have symbols?   Not sure. Although honestly I'm not sure that we'd analyze crash dumps in this scenario. We might just run VS locally (via experimental instance) to reproduce.

What do you think?

Comment (dtivel) [reply eng/pipelines/templates/Build_and_UnitTest.yml:478]: My guess is that publishing on every build improved the debugging experience for all official builds --- at least for 45 days ---, irrespective of whether or not the build was inserted.  Sometimes people test official builds before they may be inserted.  Unless there's a problem, it seems like a good rule to keep.

This change is at least [3 years old](https://github.com/NuGet/NuGet.Client/commit/7537e3a59c211c4896056a2777dec2a79577b28e#diff-6642474c9c1e8dc7f5c5ed2c99c67b4c086580b80032219592c2df3f44f86026R243).  I don't remember details.

Comment (nkolev92) [reply eng/pipelines/templates/Build_and_UnitTest.yml:478]: I don't have any recollection as to why this was added either.

Comment (zivkan) [reply eng/pipelines/templates/Build_and_UnitTest.yml:478]: The `artifactSymbolTask` task is using a deprecated API and has been outputting warnings telling us (well, the task maintainer) to migrate to another task, but we never noticed since we don't look at the log. I got feedback from someone in the MicroBuild team to use the official AzDO "index source and pusblish symbol" task. but it doesn't have an expiration parameter.

Additionally, this `artifactSymbolTask` task doesn't work on MicroBuild's scale set agent pool that we're required to move to, so there is a problem.

Since there's only a hypothetical (who is debugging official builds not inserted?) benefit and it's less effort to delete than to figure out the repercussions of switching to a different task that doesn't have the same feature, I still propose we delete it.

Comment (dominoFire) [reply eng/pipelines/templates/Build_and_UnitTest.yml:478]: Where do symbols from official NuGet builds are published ? If you remove this task, I believe we will not publish symbols for official builds.

Should we migrate to [AzDO task to publish symbols](https://docs.microsoft.com/en-us/azure/devops/pipelines/tasks/build/index-sources-publish-symbols?view=azure-devops), as [documented here](https://www.1eswiki.com/wiki/Using_Azure_DevOps_Symbols#Publishing_Symbols)?

File: eng/pipelines/templates/Build_and_UnitTest.yml
Comment (dominoFire) [inline eng/pipelines/templates/Build_and_UnitTest.yml:470]: Would be nice to update all VSTS naming to Azure DevOps or AzDO

```suggestion
  displayName: "Upload AzDO Drop"
```

**Review by dominoFire — APPROVED** [review]
Sign-off to unblock.

File: eng/pipelines/templates/Build_and_UnitTest.yml
Comment (zivkan) [reply eng/pipelines/templates/Build_and_UnitTest.yml:478]: Our pipeline to insert NuGet into VS publishes symbols. As you can see from the task that I deleted here, it was only publishing for 45 days, which would not be suitable for builds inserted into VS or the .NET SDK anyway.

File: eng/pipelines/templates/pipeline.yml
Comment (erdembayar) [inline eng/pipelines/templates/pipeline.yml:63]: Where does this pool name coming from? Is it name of VS build machine somewhere or it's something we can decide?

File: eng/pipelines/templates/Build_and_UnitTest.yml
Comment (erdembayar) [inline eng/pipelines/templates/Build_and_UnitTest.yml:50]: Does this `@6` have any impact on build? Looks like some numbering. In general can we give any name appropriate for the context or it's already decided somewhere else?

File: eng/pipelines/templates/Build_and_UnitTest.yml
Comment (zivkan) [reply eng/pipelines/templates/Build_and_UnitTest.yml:478]: Ah, the 1ES docs say something that the AzDO docs don't (maybe because it's specific to microsoft.artifacts.visualstudio.com?). Publishing there does indeed have an expiration date, so it is directly equivalent to  the old, unmainted, task. I'll update and run another build to test.

File: eng/pipelines/templates/Build_and_UnitTest.yml
Comment (zivkan) [reply eng/pipelines/templates/Build_and_UnitTest.yml:50]: https://docs.microsoft.com/en-us/azure/devops/pipelines/yaml-schema?view=azure-devops&tabs=schema%2Cparameter-schema#task

> `- task: string  # reference to a task and version, e.g. "VSBuild@1"`

File: eng/pipelines/templates/pipeline.yml
Comment (zivkan) [reply eng/pipelines/templates/pipeline.yml:63]: The first two steps of this docs page shows how to see all the agent pools: https://docs.microsoft.com/en-us/azure/devops/pipelines/agents/pools-queues?view=azure-devops&tabs=yaml%2Cbrowser

Basically, we need to stay in communication with our Engineering Systems teams to find out what infrastructure they provide for us.

File: eng/pipelines/templates/Build_and_UnitTest.yml
Comment (zivkan) [reply eng/pipelines/templates/Build_and_UnitTest.yml:478]: Reading the docs more carefully, this is something that will be easier in the pipeline that runs after the build, rather than during the build. So again, I propose cutting it out of the build, as this PR is currently doing.

File: eng/pipelines/templates/Build_and_UnitTest.yml
Comment (zivkan) [reply eng/pipelines/templates/Build_and_UnitTest.yml:470]: I do a lot of CI YAML changes, so I'll remember to change it in another commit.

Comment (nkolev92) [reply eng/pipelines/templates/Build_and_UnitTest.yml:470]: fwiw, I'd avoid AzDO/VSTS branding mentions if possible :) That way it's not a moving target.

**Pull request merged**
